packages/powersync/lib/src/database/web/web_powersync_database.dart

```diff
@@ -189,22 +191,22 @@ class PowerSyncDatabaseImpl
   /// Uses the database writeTransaction instead of the locally
   /// scoped writeLock. This is to allow the Database transaction
   /// tracking to be correctly configured.
-  Future<T> writeTransaction<T>(
-      Future<T> Function(SqliteWriteContext tx) callback,
-      {Duration? lockTimeout,
-      String? debugContext}) async {
-    await isInitialized;
-    return database.writeTransaction(
-        (context) => internalTrackedWrite(context, callback),
-        lockTimeout: lockTimeout);
-  }
+  // Future<T> writeTransaction<T>(
```

Comment:
The `writeTransaction` method on the web PowerSync database client had an override to avoid using the default implementation provided by the `SqliteQueries` [mixin](https://github.com/powersync-ja/sqlite_async.dart/blob/9428ad9ec49f0a12a5c0bf351b6bc56b359f99bb/packages/sqlite_async/lib/src/sqlite_queries.dart#L111) from `sqlite_async`. If the mixin was used it would start a `writeLock` on the `database` and then manually start a transaction.

The reason for the override is that the underlaying web `database` has custom functions for transaction SQL execution in it's `writeTransaction` method. We want to use that logic on the PowerSyncDatabase level - so we avoid the mixin implementation and call the underlaying `database` writeTransaction method instread.

Reply:
The concern I have with this is with the new custom sqlite wasm that has the rust methods, will it not execute `UPDATE ps_tx SET current_tx = next_tx, next_tx = next_tx + 1 WHERE id = 1` in  the `internalTrackedWrite()` function?

Reply:
My understanding is that the rust core intercepts [begin](https://github.com/powersync-ja/powersync-sqlite-core/blob/5704c7fb689b84f8679b006023f3fd0858fd75ff/crates/core/src/crud_vtab.rs#L171) events and automatically tracks those transactions. I don't think we are creating the transaction multiple times - so I don't think anything should be tracked twice.

Reply:
That is correct - the rust core now handles that (and only increments the transaction number when actually writing to ps_crud).